<launch>
  <arg name="device"        default="/dev/serial/by-id/usb-BestTechnology_E148_E148-if00-port0"   />
  <arg name="baud"          default="57600"          />
  <arg name="motor_count"   default="2"              />
  <arg name="freq"          default="20"             />

  <arg name="goal_topic"    default="/sensor/motor/input/position"      />
  <arg name="pos_topic"     default="/sensor/motor/output/position"   />
  <arg name="cur_topic"     default="/sensor/motor/output/current"    />

  <!-- arduino -->
  <node pkg="rosserial_python" type="serial_node.py" name="arduino_serial_node" output="screen">
    <param name="port" value="/dev/serial/by-id/usb-Arduino__www.arduino.cc__Arduino_Due_Prog._Port_2423831343535160F0C1-if00" />
    <param name="baud" value="115200" />
  </node>

  <!-- dynamixel -->
  <node pkg="dynamixel_control" type="multi_pos_ctrl_node" name="multi_pos_ctrl_node" output="screen">
    <param name="device_name"  value="$(arg device)"        />
    <param name="baud_rate"    value="$(arg baud)"          />
    <param name="motor_count"  value="$(arg motor_count)"   />
    <param name="freq"         value="$(arg freq)"          />
    <remap from="/goal_positions"      to="$(arg goal_topic)" />
    <remap from="/present_positions"   to="$(arg pos_topic)"  />
    <remap from="/present_currents"    to="$(arg cur_topic)"  />
  </node>

  <node pkg="bilateral_test" type="mag_tension_converter" name="mag_tension_converter" output="screen">
    <rosparam file="$(find bilateral_test)/config/csv_params.yaml" command="load" />
  </node>

  <node name="pd_control_node" pkg="feedback_control" type="pd_control_node" output="screen">
    <!-- PD制御のパラメータ -->
    <param name="Kp" value="20.0" />
    <param name="Kd" value="0.1" />
    <param name="publish_index" value="0" />
    <!-- モータのi番目が磁気センサの何番目か？ -->
    <param name="tension_id" value="{2, 0}" />
    <param name="motor_inverse" value="{-1, 1}" />
  </node>
</launch>